### PR TITLE
[14.0][MIG] animal

### PR DESCRIPTION
**Pull request by josarmas9416**
Module migrated to Odoo 14.0 and fully translated to Spanish.

### CODE REVIEW

**Comment by fshah-initos** [comment]
Code Review Done and Functionality Tested.
LGTM

**Comment by kos94ok-3D** [comment]
@josarmas9416, please squash three last commits

**Comment by josarmas9416** [comment]
> @josarmas9416, please squash three last commits

Sorry about that, It's done.

**Review by fshah-initos — APPROVED** [review]

**Comment by fshah-initos** [comment]
Hi @kos94ok-3D ,
@josarmas9416 has squash the commits can you please have look and add the reviews here?

**Review by kos94ok-3D — APPROVED** [review]

File: animal/__init__.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Copyright (C) 2020 Open Source Integrators
```
Comment (NL66278) [inline animal/__init__.py]: Please remove copyright lines from init files. Also make license lines https: (in all files).

File: animal/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        "data/animal.color.csv",
+        "security/res_groups.xml",
+        "security/ir.model.access.csv",
+        "views/animal_color.xml",
+        "views/animal_breed.xml",
+        "views/animal_species.xml",
+        "views/animal.xml",
+        "views/menu.xml",
+    ],
+    "demo": [],
```
Comment (NL66278) [inline animal/__manifest__.py]: Leave out empty keys. Although maybe some of the data should really be in demo? Why only dog breeds? Why the particular selection of colors and species?

File: animal/models/animal.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    active = fields.Boolean(default=True)
+    image = fields.Binary(
+        "Image", attachment=True, help="This field holds the photo of the animal."
+    )
+
+    @api.onchange("species_id")
+    def onchange_species(self):
+        self.breed_id = False
+
+    @api.onchange("breed_id")
```
Comment (NL66278) [inline animal/models/animal.py:40]: Why? If you have a white dog with black spots and you decide it is not an Dalmation after all, does that change the color?

File: animal/models/animal_color.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# Copyright (C) 2020 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class AnimalColor(models.Model):
```
Comment (NL66278) [inline animal/models/animal_color.py:6]: Why is color linked to species and breed? Black is black, or not?

File: animal/views/animal.xml [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+                <kanban class="o_res_partner_kanban">
+                        <field name="id" />
+                        <field name="image" />
+                        <field name="name" />
+                        <templates>
+                            <t t-name="kanban-box">
+                                <div
+                                class="oe_kanban_global_click o_kanban_record_has_image_fill o_res_partner_kanban"
+                            >
+                                    <t t-if="record.image.raw_value">
```
Comment (NL66278) [inline animal/views/animal.xml]: Too much indented?

**Comment by fshah-initos** [comment]
Hi @josarmas9416 can you please have a look at comments so that we can merge this PR soon.

**Comment by josarmas9416** [comment]
> Hi @josarmas9416 can you please have a look at comments so that we can merge this PR soon.

Hi, I just migrated the module as it was from version 12. should making these required changes in functionality be a separate issue or am I wrong?

**Comment by fshah-initos** [comment]
Some of the comments are not related to the functionality you cna resolve that.

**Comment by NL66278** [comment]
@josarmas9416 I agree with @fshah-initos . You can keep the functionality as is. Although would be nice if authors of the module answered the questions, or even better put some rationale in the README. But please attent to the comments on formatting and conventions.

**Comment by fshah-initos** [comment]
Hi @josarmas9416 It would be great if you could have a look at the comments.

**Review by fshah-initos — APPROVED** [review]
LGTM

**Comment by fshah-initos** [comment]
Maintainers can you please merge this PR?

**Review by NL66278 — APPROVED** [review]
:+1: LGTM I have some questions on the design of this module that are unanaswered, like why are breed and color linked, but as the module users have apparently no problem with this I approve.

**Pull request merged**